src/Symfony/Component/Serializer/Tests/Normalizer/AbstractNormalizerTest.php

```diff
@@ -132,6 +132,14 @@ public function testObjectWithNullableConstructorArgument()
         $this->assertNull($dummy->getFoo());
     }
 
+    public function testObjectWithNullableNonOptionalConstructorArgument()
```

Comment:
`NullableNonOptional` = `RequiredNullable` which ideally is also conveyed by `NullableConstructorArgumentDummy` (thus `RequiredNullableConstructorArgumentDummy`). As it potentially can invalidate this test.

Perhaps if the rename is a hassle, add a class_alias instead?

Reply:
That's true, I hesitated to create a new dummy class for the test, but since the other one was exactly the same I decided not to copy it, but I can add it if it makes sense.

Reply:
I wouldn't use `class_alias` for this, it's a little magic that can end up as a big WTF for debugging.

Reply:
we could try updating the signature of the current dummy class to see what tests do (eg. `= null), if all good then add a new `RequiredNullableConstructorArgumentDummy` specifically for this test.

Reply:
How about I add a class whose parameter has a default value of null, and I duplicate both tests so that each one is testing both classes to complete the test matrix ? That way all use case would be covered.

Reply:
alternatively `?string $required, ?string $optional = null` in the current fixture may work :)

Reply:
Yes, probably, it'll do less tests, but would be less explicit. I don't know if there's any guideline about that in Symfony, I personally am more comfortable with unit tests (unit as in atomic unit) for this kind of use case, so if no one steps up and ask me to do otherwise, I'll do the 4 tests / 2 classes variants :)

Reply:
Done !